vcsh

```diff
@@ -234,17 +235,20 @@ foreach() {
 	# We default to prefixing `git` to all commands passed to foreach, but
 	# allow running in general context with -g
 	command_prefix=git
-	while getopts "g" flag; do
+	command_suffix='cat'
+	while getopts "gp" flag; do
 		if [ x"$1" = x'-g' ]; then
 			unset command_prefix
 		fi
+		if [ x"$1" = x'-p' ]; then
+			command_suffix='sed "s/^/$VCSH_REPO_NAME: /"'
+		fi
 		shift 1
```

Comment:
Does that DTRT when argv is `-gp` as one word?

Reply:
No, and it's a limitation of the `shift 1` approach.

The getopts $flag variable should be parsed and then handled with a case/esac.